tensorboard/components/tensor_widget/tensor-widget-impl.ts

```diff
@@ -521,16 +623,152 @@ export class TensorWidgetImpl implements TensorWidget {
             i < (values as number[][]).length &&
             j < (values as number[][])[i].length
           ) {
+            const value = (values as number[][])[i][j];
             valueDiv.textContent = numericValueToString(
-              (values as number[][])[i][j],
+              value,
               isIntegerDType(this.tensorView.spec.dtype)
             );
+            valueDiv.setAttribute(DETAILED_VALUE_ATTR_KEY, String(value));
           } else {
             valueDiv.textContent = '';
+            valueDiv.setAttribute(DETAILED_VALUE_ATTR_KEY, '');
           }
         }
       }
     }
+    this.renderSelection();
+  }
+
+  /**
+   * Update the rendering of the selected value cells (if any).
+   */
+  private renderSelection() {
+    if (this.selection == null) {
+      return;
+    }
+    const numRows = this.valueDivs.length;
+    const numCols = this.valueDivs[0].length;
+    for (let i = 0; i < numRows; ++i) {
+      for (let j = 0; j < numCols; ++j) {
+        const valueDiv = this.valueDivs[i][j];
+        valueDiv.classList.remove('tensor-widget-value-div-selection');
+        valueDiv.classList.remove('tensor-widget-value-div-selection-top');
+        valueDiv.classList.remove('tensor-widget-value-div-selection-bottom');
+        valueDiv.classList.remove('tensor-widget-value-div-selection-left');
+        valueDiv.classList.remove('tensor-widget-value-div-selection-right');
+        const indices = this.calculateIndices(i, j);
+        const status = this.selection.getElementStatus(indices);
+        if (status !== null) {
+          valueDiv.classList.add('tensor-widget-value-div-selection');
+          if (status.topEdge) {
+            valueDiv.classList.add('tensor-widget-value-div-selection-top');
+          }
+          if (status.bottomEdge) {
+            valueDiv.classList.add('tensor-widget-value-div-selection-bottom');
+          }
+          if (status.leftEdge) {
+            valueDiv.classList.add('tensor-widget-value-div-selection-left');
+          }
+          if (status.rightEdge) {
+            valueDiv.classList.add('tensor-widget-value-div-selection-right');
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Calcualte the set of indices that a value div currently maps to.
+   * @param row Row index of the value div, 0-based. This is with respect
+   *   to the 2D array of value divs that the widget currently possess, not
+   *   with respect to the indices of the underlying tensor. Same below.
+   * @param col Column indices of the value div, 0-based.
+   * @return The set of indices of the underlying tensor.
+   */
+  private calculateIndices(row: number, col: number): number[] {
```

Comment:
How about `viewRow` (vs. `tensorRow`), and similarly `calculateTensorIndices()`, to make that distinction as clear as possible